carma-messenger-core/cpp_message/src/MobilityOperation_Message.cpp

```diff
@@ -159,72 +157,116 @@ namespace cpp_message
+            ROS_WARN("Unacceptable GUID, changing to default");
+            plan_id=Header.GUID_DEFAULT;
+            string_size=Header.GUID_DEFAULT.size();
+        }
         uint8_t string_content_planId[string_size];
         for(size_t i=0;i<string_size;i++)
         {
-            string_content_planId[i]=plainMessage.header.plan_id[i];
+            string_content_planId[i]=plan_id[i];
         }
         message->value.choice.TestMessage03.header.planId.buf=string_content_planId;
         message->value.choice.TestMessage03.header.planId.size=string_size;
         //get timestamp and convert to char array
         uint64_t time=plainMessage.header.timestamp;
         std::string timestamp=std::to_string(time);
         string_size=timestamp.size();
-        uint8_t string_content_timestamp[string_size];
-        for(size_t i=0;i<string_size;i++)
+        if(string_size<Header.TIMESTAMP_MESSAGE_LENGTH){
+            timestamp=std::string((Header.TIMESTAMP_MESSAGE_LENGTH-string_size),'0').append(timestamp);
+        }
+        else if(string_size>Header.TIMESTAMP_MESSAGE_LENGTH){
+            ROS_WARN("Unacceptable timestamp value, changing to default");
+            timestamp=std::string(Header.TIMESTAMP_MESSAGE_LENGTH,'0');
+        }
+        uint8_t string_content_timestamp[Header.TIMESTAMP_MESSAGE_LENGTH];
+        for(size_t i=0;i<Header.TIMESTAMP_MESSAGE_LENGTH;i++)
         {
             string_content_timestamp[i]=timestamp[i];
         }
         message->value.choice.TestMessage03.header.timestamp.buf=string_content_timestamp;
-        message->value.choice.TestMessage03.header.timestamp.size=string_size;
+        message->value.choice.TestMessage03.header.timestamp.size=Header.TIMESTAMP_MESSAGE_LENGTH;
 
         //convert strategy string to char array
-        string_size=plainMessage.strategy.size();
+        std::string strategy=plainMessage.strategy;
+        string_size=strategy.size();
+        if(string_size<STRATEGY_MIN_LENGTH || string_size>STRATEGY_MAX_LENGTH){
+            ROS_WARN("Unacceptable strategy_params value, changing to default");
+            strategy=Header.STRING_DEFAULT;
+            string_size=Header.STRING_DEFAULT.size();
+        }        
         uint8_t string_content_strategy[string_size];
         for(size_t i=0;i<string_size;i++)
         {
-            string_content_strategy[i]=plainMessage.strategy[i];
+            string_content_strategy[i]=strategy[i];
         }
         message->value.choice.TestMessage03.body.strategy.buf=string_content_strategy;
         message->value.choice.TestMessage03.body.strategy.size=string_size;
         
 
         //convert parameters string to char array
-        string_size=plainMessage.strategy_params.size();
+        std::string strategy_params=plainMessage.strategy_params;
+        string_size=strategy_params.size();
+        if(string_size<STRATEGY_PARAMS_MIN_LENGTH || string_size>STRATEGY_PARAMS_MAX_LENGTH){
```

Comment:
Is there a reason not to pad this to min length? I guess because it's not a stringified number there's not a super clean way to do it? Doesn't have to change, just trying to understand the rationale.

Reply:
Minimum length here is 2, I could pad it, but would it make sense in that case?

Reply:
Actually, looking at it this is a fine resolution, since the default string is "[]" anyway, which is fine since the brackets are required regardless.